src/main/java/duke/Main.java

```diff
@@ -0,0 +1,140 @@
+package duke;
+
+import java.time.LocalDate;
+
+import duke.command.CommandParser;
+import duke.command.CommandType;
+import duke.commons.exceptions.DukeException;
+import duke.storage.PersistentStorageHandler;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.TaskList;
+import duke.task.ToDo;
+import duke.ui.UserInterface;
+
+public class Main {
+
+    private static TaskList taskList = new TaskList();
+
+    private static PersistentStorageHandler persistentStorageHandler = new PersistentStorageHandler();
+
+    public static void main(String[] args) {
+
+        UserInterface.printWelcome();
+
+        try {
+            if (persistentStorageHandler.ensureTaskFileExists()) {
+                taskList = persistentStorageHandler.readTaskFileFromDisc();
+                int numTasks = taskList.getNumberTasks();
+                UserInterface.print("Read existing tasks (" + numTasks + ") from disc");
+            }
+        } catch (DukeException e) {
+            UserInterface.showError(e.getMessage());
+        }
+
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String userInput = UserInterface.getUserInput();
+                CommandType commandType = CommandParser.parseCommand(userInput);
+
+                switch (commandType) {
```

Comment:
Please change the indentation of switch and case statements to follow the standards of this mod! 🗿